fzf-git.sh

```diff
@@ -132,7 +132,7 @@ fi
 _fzf_git_files() {
   _fzf_git_check || return
   (git -c color.status=always status --short
-   git ls-files | grep -vf <(git status -s | grep '^[^?]' | cut -c4-) | sed 's/^/   /') |
+   git ls-files | (excludes="$(git status -s | grep '^[^?]' | cut -c4-)"; [ -n "${excludes}" ] && grep -vf <(echo "${excludes}") || cat) | sed 's/^/   /') |
```

Comment:
```suggestion
   git ls-files | grep -vxFf <(git status -s | grep '^[^?]' | cut -c4-; echo $'\0') | sed 's/^/   /') |
```

Can you check if this helps? Print a NUL byte so that the file grep reads is no longer empty, but it's a NUL byte so it won't match any file.

Reply:
> Can you check if this helps? Print a NULL character so that the file grep reads is no longer empty, but it's a NULL character so it won't match any file.

I'm getting a different error in that case:
```
$ echo a | grep -vxFf <(git status -s | grep '^[^?]' | cut -c4-; echo $'\0')
grep: empty (sub)expression
```

Same thing with an empty string
```
$ echo a | grep -vxFf <(git status -s | grep '^[^?]' | cut -c4-; echo '')
```

A compromise could be to add `echo '^$'` which would filter out empty lines
```
$ echo a | grep -vxFf <(git status -s | grep '^[^?]' | cut -c4-; echo '^$')
a
```

> I think we should just tell the users running old macOS to install GNU grep (`brew install grep`) instead of special-casing each invocation of it, then the script can pick up `ggrep` if it's found.

This is perfectly acceptable imo.

Reply:
> A compromise could be to add `echo '^$'` which would filter out empty lines

I like this. But because of `-F` I added above, `^$` will be recognized as the literal `^$`. It will work because there will be no file named as such, but the code can be a bit confusing. So how about if we echo `:`, the common path list separator. It is extremely unlikely that there is a file named `:` in a repository.

```suggestion
   git ls-files | grep -vxFf <(git status -s | grep '^[^?]' | cut -c4-; echo :) | sed 's/^/   /') |
```